packages/web-react/src/types/shared/inputs.ts

```diff
@@ -0,0 +1,31 @@
+export type ValidationState = 'error';
+
+export interface Validation {
+  /** Whether the input should display its "valid" or "invalid" visual styling. */
+  validationState?: ValidationState;
+  /**
+   * Whether user input is required on the input before form submission.
+   */
+  required?: boolean;
+}
+
+export interface InputBase {
```

Comment:
Why not `id` as well?

Reply:
Because this is not general `id` but a value used for `htmlFor` prop, e. g. `id` + `for`. And not every input must have an `id`.